src/opi/input/structures/properties.py

```diff
@@ -0,0 +1,321 @@
+
+        Parameters
+        ----------
+        xyz_lines: TrackingTextIO
+            A buffer that contains xyz file data
+        comment_symbols: str | Sequence[str] | None, default: None
+            List of symbols that indicate user comments in the xyz file. User comments are skipped before the actual xyz
+            data starts. By default, no user comments are used. White-space only comments are not allowed and are
+            silently ignored.
+
+        Raises
+        --------
+        ValueError
+            When no valid properties can be read from the input buffer or the corresponding structure is incomplete
+
+        Returns
+        --------
+        Properties | None
+            The `Properties` object extracted from the buffer or None if the buffer was empty.
+        """
+        # > Select mode
+        mode_functions = {
+            "goat": cls.goat_energies,
+            "docker": cls.docker_energies,
+        }
+
+        comments_tuple: tuple[str, ...] | None = None
+
+        # > Convert comments to tuple
+        if isinstance(comment_symbols, str):
+            comments_tuple = (comment_symbols,)
+        elif isinstance(comment_symbols, Sequence):
+            comments_tuple = tuple(comment_symbols)
+
+        # > Skip arbitrary number of empty and user comment lines at the beginning
+        while (line := xyz_lines.readline()) != "":
+            if not line.lstrip():
+                continue
+            # > Check for comment line. Ignore empty/whitespace lines
+            elif comments_tuple and line.lstrip().startswith(comments_tuple):
+                continue
+            else:
+                break
+
+        # > No data available in the buffer
+        if line == "":
+            return None
+
+        # > Fetch number of atoms
+        try:
+            natoms = int(line.split()[0])
+        except (ValueError, IndexError) as err:
+            raise ValueError(
+                f"Line {xyz_lines.line_number}: Could not read number of atoms at the beginning of xyz data"
+            ) from err
+
+        # > Comment line
+        line = xyz_lines.readline()
+        if line == "":
+            raise ValueError(
+                f"Line {xyz_lines.line_number}: Comment line is not present in xyz data"
+            )
+
+        # > Analyse comment line
+        properties = mode_functions[mode](line)
+
+        # > Skip the remaining structure
+        for iline in range(natoms):
+            line = xyz_lines.readline()
+            # > empty lines are not allowed
+            if line == "":
+                raise ValueError(f"Line {xyz_lines.line_number}: Incomplete xyz file buffer")
+
+        return properties
+
+    @classmethod
+    def docker_energies(cls, line: str) -> "Properties":
+        """Function for reading docker energies from string and return a Properties object."""
```

Comment:
1)
```suggestion
        """Function for reading DOCKER energies from string and return a Properties object."""
```

2)
From any string? Please document what the string has to look like.